linkerd/app/core/src/errors.rs

```diff
@@ -265,12 +265,15 @@ fn set_grpc_status(
         headers.insert(GRPC_STATUS, code_header(code));
         headers.insert(GRPC_MESSAGE, HeaderValue::from_static("request timed out"));
         code
-    } else if error.is::<FailFastError>() {
+    } else if let Some(e) = error.downcast_ref::<FailFastError>() {
         let code = Code::Unavailable;
         headers.insert(GRPC_STATUS, code_header(code));
         headers.insert(
             GRPC_MESSAGE,
-            HeaderValue::from_static("proxy max-concurrency exhausted"),
+            HeaderValue::from_str(&e.to_string()).unwrap_or_else(|error| {
```

Comment:
since the failfast scopes are static strings, i wonder if we could just add more of the message to them, so that they can be used with `HeaderValue::from_static`...string overhead probably doesn't matter that much so this is probably unnecessary...

Reply:
Yeah, probably more we could yakshave around this; but later.